### DISCUSSION TITLE
Permissions for files, ability to make them public or not

### DISCUSSION THREAD

**Discussion by epenabella** [directus/directus#10920] [Archive]
Hopefully this hasn't been brought up already somewhere but i did do an initial scan

**Comment by azrikahar** [comment]
You can do so by [giving Read permissions (or other permissions if needed) for Public role](https://docs.directus.io/configuration/users-roles-permissions/#configuring-permissions) to the `directus_files` system collection:

https://user-images.githubusercontent.com/42867097/148645149-46ed285a-cb9a-4814-9474-bfa07a82f295.mp4

However you may want to avoid opening up _all_ files to the public for security reasons. One way to do so would be to only allow permission to files _within_ a specific folder. You can do that by using [filter rules](https://docs.directus.io/configuration/filter-rules/) for more fine grained control:

https://user-images.githubusercontent.com/42867097/148645238-0315f797-74de-4c2b-890e-e0a1b6c6c48b.mp4

**Comment by erondpowell** [comment]
Tangentially related question. When I use REST to GET the items from the `directus_files` system collection, it returns a Forbidden error, even when I add an admin static token. Is that normal? I'm doing this after Thoth asked about height/width on directus_files. 
https://discord.com/channels/725371605378924594/741330910594465823/971459386578964500
The Collection has Fields for this _(it actually has fields for a lot of stuff)_ but they are not displayed in the File Library Module.

BTW, here is what happens when I try to get the metadata (items) from `directus_files`.
I added a dummy static token to my admin account to try and make this work, but to no avail.

https://user-images.githubusercontent.com/62306630/166826733-849d8beb-de4b-46cd-a92f-0ea5425cdaed.mov

**Reply by azrikahar** [reply]
It should be `/files` instead of `/items/directus_files` (true for system collections, such as Directus Collections should be `/collections` instead of `/items/directus_collections`) 👍 Can you try that instead?

As a bit of extra info, the reason why it's showing as forbidden should be because ItemsService specifically checks whether it starts with `directus_` and returns Forbidden here:

https://github.com/directus/directus/blob/1ec4fe165b674c413efc0ee708f2425191e0f6d1/api/src/controllers/items.ts#L58-L59

**Reply by erondpowell** [reply]
@azri, this works! `/files` returns the Items from the `directus_files` Collection.
Link to the docs for anybody else who needs it: https://docs.directus.io/reference/files/
